### ISSUE TITLE
Input clef/key sig/time sig before cursor in note input mode

### ISSUE THREAD

**Issue by oktophonie** [musescore/MuseScore#24763]
### Your idea

While in note input mode, a clef, key signature or time signature can be added from the palette. However, these items will be added not the point where the cursor currently is, but on the currently selected (i.e. last-inputted) note.

It would be more intuitive if these items were added at the cursor position.

### Problem to be solved

For example: enter note input mode and enter these notes by typing D C B A:
![image](https://github.com/user-attachments/assets/aae50a93-d833-4080-8309-0a1d722c7b4e)

Let's say I want a clef change before the next note. If I click one from the palette now, it will be inserted before the _selected_ note (the A) - just as would happen outside of note input mode:
![image](https://github.com/user-attachments/assets/04de3524-db33-44c4-91bf-2ea22248e5dc)

Since we're entering items sequentially it would make more sense (to me, anyway) to input it at the cursor position, so we'd get this:
![image](https://github.com/user-attachments/assets/76dead21-0da8-492b-aa7e-bf8d6bcede27)

As it is, you have to input the next note _then_ input the clef, which feels a bit backwards.

Same goes for key/time sigs, which in this case are added to the previous bar, rather than the one where the cursor is.

### Checklist

- [X] This request follows the [guidelines](https://github.com/musescore/MuseScore/wiki/Reporting-bugs-and-issues) for reporting issues
- [X] I have verified that this feature request has not been logged before, by searching the [issue tracker](https://github.com/musescore/MuseScore/issues) for similar requests

**Comment by wizofaus** [comment]
The commands "Add clef: bass" and "Add clef: treble" do work like this, even if it seems to require defining a custom keyboard shortcut for them.
In general I find there's a bit of a UX issue with how to specify that you want something to apply to a note just entered, vs the following note(s), as often both are useful operations.

**Comment by sammik** [comment]
see #24811

**Comment by wizofaus** [comment]
Ok, but now how do you retrospectively add a clef change before the note just entered because you've realised it is further off the staff than you expected?

**Comment by wizofaus** [comment]
(my personal preference is just to use the left arrow key to move the current position back before applying a change to an already entered note, but obviously needs agreement from those in charge of such decisions!)

**Comment by sammik** [comment]
> (my personal preference is just to use the left arrow key to move the current position back before applying a change to an already entered note, but obviously needs agreement from those in charge of such decisions!)

exactly, as issue title says "Input clef/key sig/time sig before **cursor** in note input mode"

Before cursor, wherever it is.

**Comment by Jojo-Schmitz** [comment]
Seems you're PR adds the clef before the next note, not after the last, resulting in the clef to be the first thing in the next measure rather than the last in the previous

**Comment by sammik** [comment]
> Seems you're PR adds the clef before the next note, not after the last, resulting in the clef to be the first thing in the next measure rather than the last in the previous, so not what @oktophonie's image suggests?

I recheck it. Working as expected (Simons third image).

**Comment by Jojo-Schmitz** [comment]
You're right, it's my backport that does not
